### ISSUE TITLE
[$250] Error message appears when requesting IOU to a secondary login of a phone number

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37835]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**  secondary email([dbarrett@quinthar.com](mailto:dbarrett@quinthar.com)) of a SMS number account (+18018600540@expensify.sms).
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1709348197580109

## Action Performed:
1. Go to preexisting chat with secondary login email address dbarrett@quinthar.com of SMS number +18018600540@expensify.sms 
2. Request money

## Expected Result:
Able to request money

## Actual Result:
Error message appears:
`unexpected error creating this chat, please try again later. There is a previously existing chat between thest users`

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![image](https://github.com/Expensify/App/assets/38435837/6c959cbe-ef56-4c3b-b312-0a05df427344)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0189bc700718cf5a85</li>
        <li>Upwork Job ID: 1766165401103810560</li>
        <li>Last Price Increase: 2024-03-15</li>
    </ul>
</details>

**Comment by twisterdotcom** [comment]
I can't recreate this:

https://github.com/Expensify/App/assets/9133401/3f099d8b-a67f-4bd9-b0db-972317369913

@mallenexpensify laid out some steps for `Needs Reproduction` issues in Slack: https://expensify.slack.com/archives/C01SKUP7QR0/p1709232158079079?thread_ts=1709232107.784279&cid=C01SKUP7QR0

> - GH created, $250 to start, bumped to $500 after 2 weeks
> - Assigned C+ reviewed discussions and steps
> - 3 contributors actively worked on finding the bug and steps
> - New issue created for $500, same C+ assigned.
> - The 3 contributors got first dibs for 48 hours to submit proposals before 'help wanted' was added.
> - Once PR is merged, ask C+ to propose the payments they recommend for contributors who helped find the repro steps.

Cool, so it sounds like we want the community to help identify the reproduction steps. Gonna wait until the outcome of this conversation so as not to share some things more widely than they need to be shared: https://expensify.slack.com/archives/C01SKUP7QR0/p1709920757503339

**Comment by c3024** [comment]
I could reproduce it by requesting money directly with the `Request money` from FAB instead of `Start chat`. It should be fixed from the backend.

### Steps:
Pre-condition: User A has an existing chat with the primary login of User B

1. [User A] Click on FAB > Request Money
2. Input amount and click `Next`
3. Input the secondary login of User B and select the result displayed below
4. Click `Request`

Expected result: A money request from User A to User B created in the existing chat between User A and B

Actual result: A new chat is created between User A and secondary login of User B with an error from backend


https://github.com/Expensify/App/assets/102477862/b83a55e2-669d-466e-a59e-225a52ccff52

**Comment by mallenexpensify** [comment]
Thanks @twisterdotcom , I created this internal issue to track, reckon it'll be a few weeks til I prioritize
- https://github.com/Expensify/Expensify/issues/377295

I think the first step, after BZ isn't able to reproduce, would be to add the `Internal` label to ask the C+ if they can and, if so, to provide additional details.  Which, for this example, would have had @c3024 able to reproduce.  Then you'd attempt again to confirm then the issue would move on. (and... sorry @c3024 , since you're C+ assigned, you wouldn't get compensation for being able to reproduce).

**Comment by twisterdotcom** [comment]
Ah nice, well okay. Let me see if I can reproduce with that way too @c3024.

**Comment by twisterdotcom** [comment]
Yeah, this is odd, the bug is real!

1. Open staging.new.expensify.com
2. Use the green plus button to Request Money > $X > dbarrett@quinthar.com
3. Witness the error.

[866608899df455ea-LHR](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%22866608899df455ea-LHR%22)+AND+timestamp:[2024-03-18T13:50:08.404Z+TO+2024-03-18T15:50:08.404Z]&index=logs_expensify-027363)
>406 Chat report with these participants already exists with different ID

[86660889ce3b55ea-LHR](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2286660889ce3b55ea-LHR%22)+AND+timestamp:[2024-03-18T13:50:08.398Z+TO+2024-03-18T15:50:08.398Z]&index=logs_expensify-027363)
>404 Report not found

https://github.com/Expensify/App/assets/9133401/3d28d66f-dbbc-4d6e-b26b-5750d8c20992

As @c3024 says, likely internal.

**Comment by twisterdotcom** [comment]
Putting it in VIP VSB.

**Comment by twisterdotcom** [comment]
It's Internal and on a WN project. Gonna keep it Weekly.

**Comment by twisterdotcom** [comment]
Asking how engineers come across these here: https://expensify.slack.com/archives/C03SSAQ7P/p1712151011245599

**Comment by twisterdotcom** [comment]
@quinthar I threw this into `Hot Picks`

**Comment by twisterdotcom** [comment]
Waiting on an internal engineer

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by twisterdotcom** [comment]
The bug still exists:
<img width="983" alt="image" src="https://github.com/Expensify/App/assets/9133401/551aff99-3944-418e-a92e-79d31ade1580">

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by twisterdotcom** [comment]
It still very much exists. @mvtglobally what flow are you testing? I get this every single time I request money/submit expense to dbarrett@quinthar.com using the FAB.

**Comment by mvtglobally** [comment]
What are we missing? 

https://github.com/Expensify/App/assets/43995119/e3da75b3-ee52-48ab-9f08-4d4896659a20


https://github.com/Expensify/App/assets/43995119/29bca1d4-0295-4638-a19f-2c7188860db5

**Comment by c3024** [comment]
@mvtglobally In the second video, you should already have chat with David's sms.

Flow in the first video does not reproduce this bug. You should start with submit expense from FAB as mentioned [here](https://github.com/Expensify/App/issues/37835#issuecomment-1987149861)

**Comment by mvtglobally** [comment]
@c3024 

Can you pls comment with the exact steps from a Brand new account. Unfortunately, we need to use new accounts as messaging David's sms is not a viable option

**Comment by twisterdotcom** [comment]
1. Create a new account
2. Create a DM chat with +18018600540@expensify.sms
3. Create a request (submit expense) to dbarrett@quinthar.com

**Comment by mvtglobally** [comment]
@twisterdotcom Are you able to reproduce it with any other account rather than [+18018600540@expensify.sms](mailto:+18018600540@expensify.sms) or [dbarrett@quinthar.com](mailto:dbarrett@quinthar.com) 

It is not reproducible if we use other SMS and email accounts

**Comment by twisterdotcom** [comment]
I literally just created an account for +447876868752 with a secondary login of twisterdotcom+sms@gmail.com and recreated the bug straight away:


https://github.com/Expensify/App/assets/9133401/ba8e4068-9ebf-4514-950b-ae8bcd3213ee



The bug is real and definitely exists.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

https://github.com/Expensify/App/assets/43995119/442f704c-72a6-44f9-bb32-7ae2aca0a244


https://github.com/Expensify/App/assets/43995119/403cb135-d125-426e-ba52-c46c7e1d52c2

**Comment by twisterdotcom** [comment]
@mvtglobally could you please test with the account I provided from above? So try creating an expense with twisterdotcom+sms@gmail.com?

**Comment by mvtglobally** [comment]
looks like we are only able to reproduce if there is an existing chat with the primary login of a user 


https://github.com/Expensify/App/assets/43995119/ffd92f3f-2792-4868-b08e-a26a0eecc5a3

**Comment by c3024** [comment]
Yes, that is what I mentioned as a pre-condition [here](https://github.com/Expensify/App/issues/37835#issuecomment-1987149861).

>Pre-condition: User A has an existing chat with the primary login of User B

**Comment by twisterdotcom** [comment]
Great, do we think this is still `Internal` now then?

**Comment by c3024** [comment]
> Great, do we think this is still `Internal` now then?

I think so.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fifth week)

**Comment by twisterdotcom** [comment]
Okay, finally I cannot reproduce. I have no idea how or when this was fixed though.
